lms/djangoapps/discussion/static/discussion/js/discussion_profile_page_factory.js

```diff
@@ -0,0 +1,29 @@
+;(function(define) {
+    'use strict';
+
+    define(['jquery', 'DiscussionUserProfileView'],
+        function($, DiscussionUserProfileView) {
+            return function(options) {
+                var element = options.el,
+                    threads = element.data('threads'),
+                    userInfo = element.data('user-info'),
+                    page = element.data('page'),
+                    numPages = element.data('num-pages');
+                // Roles are not included in user profile page, but they are not used for anything
+                window.DiscussionUtil.loadRoles({
+                    'Moderator': [],
+                    'Administrator': [],
+                    'Community TA': []
+                });
+                window.$$course_id = element.data('course-id');
```

Comment:
This is an ugly namespacing that the discussion forum code does. I just moved the logic from CoffeeScript to JavaScript but I haven't changed the implementation. Ideally the code would pass the course id around rather than storing it in a global like this. I'm going to leave it as is for now.